packages/integrations/lit/src/index.ts

```diff
@@ -0,0 +1,37 @@
+import type { AstroIntegration } from 'astro';
+
+function getViteConfiguration() {
+	return {
+		optimizeDeps: {
+			include: [
+				'@astrojs/lit/client-shim.js',
+				'@astrojs/lit/hydration-support.js',
+				'@webcomponents/template-shadowroot/template-shadowroot.js',
+				'lit/experimental-hydrate-support.js',
+			],
+			exclude: ['@astrojs/lit/server.js'],
+		},
+		ssr: {
+			external: ['lit-element/lit-element.js', '@lit-labs/ssr/lib/install-global-dom-shim.js', '@lit-labs/ssr/lib/render-lit-html.js', '@lit-labs/ssr/lib/lit-element-renderer.js'],
+		},
+	};
+}
+
+export default function (): AstroIntegration {
+	return {
+		name: '@astrojs/lit',
+		hooks: {
+			'astro:config:setup': ({ updateConfig, addRenderer, injectScript }) => {
+				injectScript('before-hydration', `import '@astrojs/lit/client-shim.js';`);
```

Comment:
This script needs to run even if there is no hydration is occurring. It polyfills an HTML feature. Maybe this should be `'head-inline'`?

Reply:
that should work, I'll check to make sure it resolves the import properly in build.

Reply:
Moved to `"head-inline"`. Note that this meant pre-bundling the polyfill since head-inline doesn't go through Vite's build. But we probably always should have been doing this, since the old "client-shim" polyfill gives no guarantee that it will execute before the code that its polyfilling (calls `await import()`) internally.